### DISCUSSION TITLE
USB connection from Android to RNode?

### DISCUSSION THREAD

**Discussion by Starfall044** [markqvist/Reticulum#227] [Help & Questions] [answered]
I realize this isn't really an "issue," maybe just a discussion question... I've read some comments in other places that suggest a direct USB connection from my phone to an RNode should work in Sideband, but other places suggest this is a feature still in development and maybe not implemented yet. In my own testing, my RNodes don't seem to connect, or Sideband doesn't see them. I just wanted to confirm before I go down a rabbit hole of troubleshooting... is that something that *should* be working?

Thanks for all the great work you've put into this remarkably comprehensive project! Google and Verizon control too much of our lives.

**Comment by markqvist** [comment]
It *is* implemented and it *should* work ;)
Try this:

- Start Sideband
- Enable connecting using RNode in the "Connectivity" screen.
- Go to the "Hardware" screen, and then to the "RNode" screen.
- Make sure radio parameters are configured.
- Just to be sure, disable "connect via bluetooth" (RNode only allows one connection at a time, if Bluetooth is *connected*, it locks out control over USB).
- Shut down Sideband (from the menu, so the background service is also closed).

Now, make sure to plug in the RNode before launching Sideband. A permission dialog should pop up asking what app to allow access to the USB device. Select Sideband. It shold start and connect to the RNode :)

**Comment by Dant3s** [comment]
It could be that the USB cable that you are using don't transport data?  Weird in 2023. But i found out using Sideband and a Heltec board that I in fact have a few cables of that type.

**Reply by markqvist** [reply]
Relevant tip! Have spent way too much time hammering my head against this problem too sometimes ;)

**Comment by Starfall044** [comment, marked as answer]
Ah, you guys are both the winners! First, I needed the more detailed
instructions from markvist, about shutting down clean from within the app
and then plugging in to have Android launch the app again. And then second:
it wasn't the cord, per se, but the little USB A-to-C adapter I was using.
I tried three of them, testing them with just USB drives, to eliminate any
other variables, and they were all power-only! But a fourth adapter, the
USB drive showed up, and then when I used that same adapter with an RNode,
presto. Thank you both!
